### DISCUSSION TITLE
/health Error

### DISCUSSION THREAD

**Discussion by thecooolemail** [vendurehq/vendure#2467] [Q&A] [answered]
Hey, 

I've managed to get vendure working on railway with a Postgres database. When using my admin dashboard every few seconds a request is sent to https://.../admin-api/health?languageCode=en which fails with a response of 

`{
    "errors": [
        {
            "message": "GraphQL operations must contain a non-empty `query` or a `persistedQuery` extension.",
            "extensions": {
                "http": {
                    "status": 400,
                    "headers": {}
                },
                "code": "BAD_REQUEST"
            }
        }
    ]
}`

And a large red box on the top right of the site appears with the same message. Im not too sure whats going on how can I resolve this error?

**Comment by michaelbromley** [comment]
Hi,
the query should normally be send to `https://domain/health` rather than `admin-api/health`. Do you have any config that might be related to this?

**Reply by thecooolemail** [reply]
Not that I know of, I dont have any files with the word "health" in it (Searched all files included in git Repo). Can I manually change the query?

<img width="509" alt="Screenshot 2023-10-20 at 23 22 06" src="https://github.com/vendure-ecommerce/vendure/assets/129298650/6f5b95a8-ffaa-4038-b3be-130a1ecdad0f">

**Reply by michaelbromley** [reply]
Is your server available publicly for me to take a look?

**Reply by thecooolemail** [reply]
Hey, I sent you an invitation to Railway and github repo to (michael@michaelbromley.co.uk) :)

**Reply by michaelbromley** [reply]
I didn't see the Railway invite. It's probably easier to debug this if you join our discord (https://vendure.io/community) and open a thread in the help channel.

**Reply by thecooolemail** [reply, marked as answer]
Removing `/admin-api` from apiHost resolved the issue.

`AdminUiPlugin.init({
            route: 'admin',
            port: 3002,
            adminUiConfig: {
                apiHost: ' https://site.up.railway.app',
            },
        }),`

**Comment by petrobenBen** [comment]
It seems like the error you're encountering is related to GraphQL and the query being sent to your admin dashboard. To resolve this issue, you can follow these steps:

Check Your GraphQL Query: Ensure that the request being sent to /admin-api/health contains a valid GraphQL query. It should be a well-formed query with the necessary fields and syntax.

Check Your Apollo Client Configuration: If you're using Apollo Client to make GraphQL requests, verify that your client is correctly configured and is including a valid query when making requests to the /admin-api/health endpoint.

Check Your Server Configuration: Make sure that your Vendure server and its GraphQL API are correctly set up to handle requests to /admin-api/health. Check if there are any misconfigurations or issues with your GraphQL schema.

Update Dependencies: Ensure that your Vendure and related packages are up to date. Outdated dependencies can sometimes lead to compatibility issues. You might want to update your Vendure version and any GraphQL-related packages.

Check for Middleware or Proxy Issues: If you have any middleware or reverse proxy (e.g., Nginx, Apache) in front of your server, make sure they are not modifying or interfering with the GraphQL requests.

Debugging: If the issue persists, use debugging tools or logging to trace the request from your admin dashboard to the server and inspect the request payload to see if there are any issues with the query being sent.

Consult Documentation/Community: Refer to the official Vendure documentation and community forums for specific solutions or to check if others have faced a similar issue.

By carefully examining your GraphQL query, client configuration, and server setup, you should be able to identify and resolve the issue. If you're still facing problems, consider providing more details about your setup for more specific assistance.

Best regards,
[petroben]

**Reply by thecooolemail** [reply]
Hey, I dont know where the request is coming from, as seen in the screen short there is no health word within the repo, Its most likely somewhere deep within one of the modules. This is a request that is made automatically every few seconds. Its not something i have added.

**Reply by petrobenBen** [reply]
[url=https://www.cosmosmagazines.com/nick-cannon-net-worth/]Nick[/url]
